source/common/http/http2/codec_impl.cc

```diff
@@ -506,14 +506,15 @@ void ConnectionImpl::sendPendingFrames() {
   }
 }
 
-void ConnectionImpl::sendSettings(uint64_t codec_options) {
+void ConnectionImpl::sendSettings(const Http2Settings& http2_settings) {
   std::vector<nghttp2_settings_entry> iv = {
-      {NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS, MAX_CONCURRENT_STREAMS},
-      {NGHTTP2_SETTINGS_INITIAL_WINDOW_SIZE, DEFAULT_WINDOW_SIZE}};
+      {NGHTTP2_SETTINGS_MAX_CONCURRENT_STREAMS, http2_settings.max_concurrent_streams_},
+      {NGHTTP2_SETTINGS_INITIAL_WINDOW_SIZE, http2_settings.initial_window_size_}};
 
-  if (codec_options & CodecOptions::NoCompression) {
+  if (http2_settings.codec_options_ &
+      Http::Http2Settings::CodecOptions::DISABLE_DYNAMIC_HPACK_TABLE) {
```

Comment:
yes, since we are using a struct now,  we can add additional fields easily.
so the flags as bool no needed any more.